fe/fe-core/src/main/java/org/apache/doris/policy/PolicyMgr.java

```diff
@@ -379,6 +389,40 @@ public RowPolicy getMatchTablePolicy(String ctlName, String dbName, String table
         return mergeRowPolicies(res);
     }
 
+    public DorisDataMaskPolicy getDataMaskPolicy(String ctlName, String dbName, String tableName, String colName, UserIdentity user) {
+        if (!hasDataMaskPolicy(ctlName, dbName, tableName, colName)) {
+            return null;
+        }
+        readLock();
+        try {
+            if (!hasDataMaskPolicy(ctlName, dbName, tableName, colName)) {
+                return null;
+            }
+            List<DorisDataMaskPolicy> dorisDataMaskPolicies = dataMaskPolicies.get(ctlName).get(dbName).get(tableName)
+                    .get(colName);
+            Set<String> roles = Env.getCurrentEnv().getAccessManager().getAuth().getRolesByUserWithLdap(user).stream()
+                    .map(role -> ClusterNamespace.getNameFromFullName(role.getRoleName())).collect(Collectors.toSet());
+
+            for (DorisDataMaskPolicy dataMaskPolicy : dorisDataMaskPolicies) {
+                if ((dataMaskPolicy.getUser() != null && dataMaskPolicy.getUser().getQualifiedUser()
```

Comment:
Unlike row policy, datamask can only return one piece of data.
So is it better to filter out users first and then find roles? This way, there can be a priority.

Reply:
Yes, there is a priority issue, which is not considered here, but only one piece of data is returned.